meson.build

```diff
@@ -4,6 +4,7 @@ project(
     version: run_command('poetry', 'version', '--short', check: true).stdout(),
     default_options: [
         'cpp_std=c++20',
+        'b_ndebug=if-release',  # disables asserts in release builds
```

Comment:
Coming from CMake (where a release build defines `NDEBUG` by default), I just *assumed* Meson would do the same. That assumption was wrong, the default is `false`. So we compiled with asserts always on. 🤡 

That immediately explains the performance issues of this PR and #319: they added a ton of assertions in really critical places. Disabling asserts in release mode should be substantially faster. I'll do a new benchmark run.

Reply:
Yeah, that fixes it.